src/lib/BLE/devBLETele.cpp

```diff
@@ -0,0 +1,262 @@
+#if defined(PLATFORM_ESP32)
+
+#include "CRSF.h"
+#include "FHSS.h"
+#include "NimBLEDevice.h"
+#include "POWERMGNT.h"
+#include "common.h"
+#include "hwTimer.h"
+#include "logging.h"
+#include "options.h"
+
+#include "devBLETele.h"
+
+#include "config.h"
+extern TxConfig config;
+
+NimBLEServer *pServer;
+NimBLECharacteristic *rcCRSF;
+
+unsigned short const TELEMETRY_SVC_UUID = 0x1819;
+unsigned short const TELEMETRY_CRSF_UUID = 0x2BBD;
+
+unsigned short const DEVICE_INFO_SVC_UUID = 0x180A;
+unsigned short const MODEL_NUMBER_SVC_UUID = 0x2A24;
+unsigned short const SERIAL_NUMBER_SVC_UUID = 0x2A25;
+unsigned short const SOFTWARE_NUMBER_SVC_UUID = 0x2A28;
+unsigned short const HARDWARE_NUMBER_SVC_UUID = 0x2A27;
+unsigned short const MANUFACTURER_NAME_SVC_UUID = 0x2A29;
+
+extern CRSF crsf;
+
+static uint32_t LastTMLLinkStatsPacketMillis = 0;
+static uint32_t LastTLMRCPacketMillis = 0;
+static bool initOnce = false;
+
+String getMasterUIDString()
+{
+    char muids[7] = {0};
+    sprintf(muids, "%02X%02X%02X", MasterUID[3], MasterUID[4], MasterUID[5]);
+    return String(muids);
+}
+
+void ICACHE_RAM_ATTR BluetoothTelemetrySendLinkStatsPacketEx(uint8_t* outBuffer)
+{
+    outBuffer[0] = CRSF_ADDRESS_RADIO_TRANSMITTER;
+    outBuffer[CRSF_TELEMETRY_LENGTH_INDEX] = CRSF_FRAME_SIZE(LinkStatisticsFrameLength);
+    outBuffer[CRSF_TELEMETRY_TYPE_INDEX] = CRSF_FRAMETYPE_LINK_STATISTICS;
+    outBuffer[CRSF_TELEMETRY_TYPE_INDEX + 1 + LinkStatisticsFrameLength] = crsf_crc.calc((byte *)&outBuffer[CRSF_TELEMETRY_TYPE_INDEX], LinkStatisticsFrameLength + 1);
+
+    rcCRSF->setValue(outBuffer, LinkStatisticsFrameLength + 4);
+    rcCRSF->notify();
+}
+
+void ICACHE_RAM_ATTR BluetoothTelemetrySendEmptyLinkStatsPacket()
+{
+    uint8_t outBuffer[LinkStatisticsFrameLength + 4];
+    memset(&outBuffer[CRSF_TELEMETRY_TYPE_INDEX + 1], 0, LinkStatisticsFrameLength);
+
+    crsfPayloadLinkstatistics_s* pStats = (crsfPayloadLinkstatistics_s*)(&outBuffer[CRSF_TELEMETRY_TYPE_INDEX + 1]);
+    pStats->uplink_RSSI_1 = 120;  
+    pStats->uplink_RSSI_2 = 120;
+    pStats->downlink_RSSI = 120;
+    pStats->uplink_SNR = -20;
+    pStats->downlink_SNR = -20;
+    pStats->rf_Mode = CRSF::LinkStatistics.rf_Mode;
+    pStats->uplink_TX_Power = CRSF::LinkStatistics.uplink_TX_Power;
+
+    BluetoothTelemetrySendLinkStatsPacketEx(outBuffer);
+}
+
+
+void ICACHE_RAM_ATTR BluetoothTelemetrySendLinkStatsPacket()
```

Comment:
```suggestion
void BluetoothTelemetrySendLinkStatsPacket()
```
Commonly used functions should not be cached in the ram, this was just me experimenting a bit.